ebtc-protocol/ebtc
# feat: Add foundry and marry it with Hardhat

Adding Foundry configuration to the project

Done:
1. Adding `nohoist` to pump openzeppelin projects to be used by Foundry in tests
2. Add foundry str-lib
3. Make sure `*.t.sol` tests are not compiled by Hardhat
4. Add couple dummy tests to make sure Forge works
5. Add GH action for Foundry runs with gas profiler

Resources:
- Guide how to match Foundry and Hardhat: https://book.getfoundry.sh/config/hardhat
Didn't work as intended. Had to add a task that excludes all `.t.sol` files from compilation


Fixes #99
